### PR TITLE
Add API version value to configuration

### PR DESCRIPTION
**Pull request by raguiar9080**
Add a new required parameter to the application that specifies the API calendar versions that is to be used. This value is currently added to the configuration of the generated client `luneClient.ts` so it can be used later. The goal is to have all requests in `lune-ts` add this value in the `Lune-Version` header so the library works on a single version, always matching the code with the types in place.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,6 +60,7 @@ $ openapi --help
     -V, --version             output the version number
     -i, --input <value>       OpenAPI specification, can be a path, url or string content (required)
     -o, --output <value>      Output directory (required)
+    -v, --apiVersions <value> API Versions currently supported separated by a comma (required)
```
Comment (rbruggem) [inline README.md:63]: is `currently` needed?

**Comment by rbruggem** [comment]
> Add a new required parameter to the application that specifies the available API versions. We are then able to generate our client taking into account these versions, allowing a new option on each endpoint to specify the version and also adding a new method to the client that allows to set the version for the client as a whole.

Are you referring to the major version that is part of the path of the date-version in the header?

**Comment by raguiar9080** [comment]
> > Add a new required parameter to the application that specifies the available API versions. We are then able to generate our client taking into account these versions, allowing a new option on each endpoint to specify the version and also adding a new method to the client that allows to set the version for the client as a whole.
> 
> Are you referring to the major version that is part of the path of the date-version in the header?

No, this refers to the calendar versions provided via header. I'll reword 👍

File: README.md (hunk 1 above)
Comment (raguiar9080) [reply README.md:63]: Nop, will reword

**Comment by raguiar9080** [comment]
Everything has been reworded. I'll wait for https://github.com/lune-climate/lune-ts/pull/665#discussion_r1745856490 to be resolved before merging since it will affect necessary changes here.

File: README.md [hunk 2]
```diff
@@ -74,8 +75,8 @@ $ openapi --help
     -h, --help                display help for command
 
   Examples
-    $ openapi --input ./spec.json --output ./generated
-    $ openapi --input ./spec.json --output ./generated --client xhr
+    $ openapi --input ./spec.json --output ./generated --apiVersions v1,v2
```
Comment (rbruggem) [inline README.md:78]: Let's use dates to make the example realistic.

File: docs/basic-usage.md [hunk 3]
```diff
@@ -23,12 +24,13 @@ $ openapi --help
     -h, --help                display help for command
 
   Examples
-    $ openapi --input ./spec.json --output ./generated
+    $ openapi --input ./spec.json --output ./generated --apiVersions v1,v2
```
Comment (rbruggem) [inline docs/basic-usage.md:27]: same as above.

**Review by rbruggem — APPROVED** [review]
Approving but I think https://github.com/lune-climate/openapi-typescript-codegen/pull/73#discussion_r1746852316 is important to avoid confusion.

**Review by rbruggem — APPROVED** [review]

**Pull request merged**